test/console/control_flow_commands_test.rb

```diff
@@ -358,4 +358,56 @@ def test_next_steps_over_rescue_when_raising_from_method
       end
     end
   end
+
+  #
+  # Tests that next/finish work for a deep call stack.
+  # We use different logic for computing frame depth when the call stack is above/below 4096.
+  #
+  if RUBY_VERSION >= '3.0.0'
```

Comment:
why it limited to the "> 3.0.0"? Too slow on the 2.7 or so?

Reply:
Yes exactly. It is too slow if we don't have the rb_profile_frames optimization in place.